### ISSUE TITLE
Inverted Preview

### ISSUE THREAD

**Issue by alyadins** [journeyapps/zxing-android-embedded#84]
Hi!
Is it possible to use inverted feature without invert preview color?

**Comment by rkistner** [comment]
There is no built-in support, but you can extend the library to allow this.

I'd guess the best place to do the inversion would be on the BinaryBitmap,
after binarization (converting to black+white) has taken place. In short,
you would need to:

1. Use barcodeView.setDecoderFactory(factory) with a custom DecoderFactory
(new class implementing DecoderFactory).
2. Let the factory create a custom Decoder (new class extending Decoder).
3. In the Decoder, override toBitmap() to invert the image after binarizing.

You would need to write the function to invert the BinaryBitmap yourself.

If you manage to get this working, I'd love to add it to the project as a
sample.

Regards,
Ralf

**Comment by rkistner** [comment]
Implemented in #235.

**Comment by romaopedro** [comment]
Implementation as @rkistner mentioned. Hope this help others :)

```
barcodeScannerView.barcodeView.decoderFactory = InvertedDecoderFactory()
```

```
class InvertedDecoderFactory : DecoderFactory {
    override fun createDecoder(baseHints: MutableMap<DecodeHintType, *>?): Decoder {
        val reader = MultiFormatReader().apply {
            setHints(baseHints)
        }
        return InvertedDecoder(reader)
    }
}
```
```
class InvertedDecoder(reader: Reader) : Decoder(reader) {
    override fun toBitmap(source: LuminanceSource): BinaryBitmap {
        return BinaryBitmap(HybridBinarizer(source.invert()))
    }
}
```
